### PR TITLE
Use refs instead of copies for small functions

### PR DESCRIPTION
**Pull request by hydroo**
Andrey and I stumbled upon that.

The (void) bvh; in isLeaf seems to cause copying (construction + destructor and its shared_pointer stuff)

What      | master | KNN   | Radius
--------- | ------ | ----- | ------
Serial    | 1      |  .949 |  .858
OpenMP 1  | 1      |  .955 |  .862
OpenMP 2  | 1      |  .955 |  .866
OpenMP 4  | 1      |  .954 |  .871
OpenMP 8  | 1      |  .959 |  .881
OpenMP 16 | 1      |  .960 |  .930
CUDA      | 1      | 1.006 | 1.005

### CODE REVIEW

**Comment by dalg24** [comment]
ok to test  (whitelisting @hydroo so that we don't have to do it again)

File: packages/Search/src/details/DTK_DetailsTreeTraversal.hpp [hunk 1]
```diff
@@ -46,12 +46,10 @@ struct TreeTraversal
      * Return true if the node is a leaf.
      */
     KOKKOS_INLINE_FUNCTION
-    static bool isLeaf( BoundingVolumeHierarchy<DeviceType> bvh,
-                        Node const *node )
+    static bool isLeaf( Node const *node )
```
Comment (dalg24) [inline packages/Search/src/details/DTK_DetailsTreeTraversal.hpp]: "Only pay for what you use" :)
Comment below might as well be removed then.

File: packages/Search/src/details/DTK_DetailsTreeTraversal.hpp [hunk 2]
```diff
@@ -60,7 +58,7 @@ struct TreeTraversal
      * Return the index of the leaf node.
      */
     KOKKOS_INLINE_FUNCTION
-    static int getIndex( BoundingVolumeHierarchy<DeviceType> bvh,
+    static int getIndex( BoundingVolumeHierarchy<DeviceType> const &bvh,
```
Comment (dalg24) [inline packages/Search/src/details/DTK_DetailsTreeTraversal.hpp]: That's right `bvh` passed here is already on the device.  Good catch!

File: packages/Search/src/details/DTK_DetailsTreeTraversal.hpp [hunk 3]
```diff
@@ -70,7 +68,7 @@ struct TreeTraversal
      * Return the root node of the BVH.
      */
     KOKKOS_INLINE_FUNCTION
-    static Node const *getRoot( BoundingVolumeHierarchy<DeviceType> bvh )
+    static Node const *getRoot( BoundingVolumeHierarchy<DeviceType> const &bvh )
```
Comment (dalg24) [inline packages/Search/src/details/DTK_DetailsTreeTraversal.hpp]: Same here, good catch.

**Comment by hydroo** [comment]
@dalg24 There are a few more copies of bvh in other functions.
Should we replace all with const & ?

File: packages/Search/src/details/DTK_DetailsTreeTraversal.hpp (hunk 2 above)
Comment (sslattery) [reply packages/Search/src/details/DTK_DetailsTreeTraversal.hpp]: nice

**Pull request merged**